libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Generators/MultiLanguageGeneratorBase.cs

```diff
@@ -40,20 +42,43 @@ public MultiLanguageGeneratorBase()
         /// <returns>The generator.</returns>
         public override async Task<string> Generate(ITurnContext turnContext, string template, object data)
         {
-            // see if we have any locales that match
             var targetLocale = turnContext.Activity.Locale?.ToLower() ?? string.Empty;
 
-            var locales = new string[] { string.Empty };
-            if (!this.LanguagePolicy.TryGetValue(targetLocale, out locales))
+            // priority 
+            // 1. local policy
+            // 2. shared policy in turnContext
+            // 3. default policy
+            var languagePolicy = this.LanguagePolicy ?? 
+                                turnContext.TurnState.Get<LanguagePolicy>() ?? 
+                                new LanguagePolicy();
```

Comment:
this you can keep, but don't monkey patch the policy, define the policy correctly.

Reply:
By "monkey patch" are you referring to override entry in policy instead of totally replacement of the whole policy?